peg_src/jsdoctype.pegjs

```diff
@@ -974,9 +974,9 @@ RecordTypeExpr = "{" _ entries:RecordTypeExprEntries? _ "}" {
                }
 
 
-RecordTypeExprEntries = first:RecordTypeExprEntry restWithComma:(_ "," _ RecordTypeExprEntry)* _ ","? {
+RecordTypeExprEntries = first:RecordTypeExprEntry restWithComma:((_ "," /_ ";" / [ \t\r ]* [\n]) _ RecordTypeExprEntry)* (_ "," / _ ";" / [ \t\r ]* [\n])? {
```

Comment:
1. I don't know pegjs syntax well enough to know: does `[ \t\r ]` include space (twice)? The documentation I found indicate that it does.
  - If so, is the intent to eat whitespace till end of line? Isn't non-newline whitespace skipped elsewere? (I expected to see `"\r"? "\n"` for parsing newlines, but maybe that won't suffice here.)
  - If not, why include `\t` in the list?
2. `[ \t\r ]*` looks like it will parse `\t\r\t\t\t\r\t`, which seems wrong to me.

Reply:
To my understanding the peg grammar just treats the contents as a regular expression character class.

The extra space was an oversight based on the fact that the earlier expression had one already (I had wrongly assumed it was some other whitespace). I've now removed it.

As far as the inclusion of tab, I was just following the previous pattern: `_ ","` where the underscore included the various whitespace characters including tab, so when I allowed `\n`, I changed the form to `_ "," /_ ";" / [ \t\r]* [\n]`, to be as equally expressive as before, but to exclude `\n` from the optional preceding whitespace which caused a parsing error (apparently due to this type of parser not being able to look ahead to match the subsequent `\n`).

I can try changing the generic whitespace to `([ \t] / ([\r]? [\n])*)*` to prevent lone `\r` markers and change the record expression to `[ \t]* ([\r]? [\n])` to do the same if that sounds good.

Reply:
It might break other stuff, but it's definitely worth a try for more understandable whitespace parsing.

Reply:
Cool, applied!